Segfault in joint trajectory controller init if joint name not in urdf

**Describe the bug**
When configuring a `joint_trajectory_controller` with joint names that are not present in the robot description/urdf, a segmentation fault is produced on initialisation. (`Segmentation fault (Address not mapped to object [0xc8])`).

The bug was introduced in #2222, which is a backport of #2163.

I rely on being able to have joint trajectory controllers for driving non-existant joints in chained mode, where the joints are not in the urdf, but rather just ros2 control abstractions being passed on to a chained controller (whose command interfaces must start with the controller's name, so as not to overwrite existing command interfaces). There's probably a more lightweight way to reproduce the issue, but this is how I was doing it.

**To Reproduce**
Steps to reproduce the behavior (this uses UR robot drivers for demonstration, but the bug is in `joint_trajectory_controller`.
1. Run `docker run -it --rm ros:jazzy`
2. In the docker container, run `apt update && apt upgrade`
3. Run `apt install ros-jazzy-ur-robot-driver`
4. Optionally run `export RCUTILS_COLORIZED_OUTPUT=1`
5. Run `ros2 launch ur_robot_driver ur_control.launch.py robot_ip:=0.0.0.0 ur_type:=ur10e use_mock_hardware:=True launch_rviz:=False` (all controllers should start up fine)
6. Modify `/opt/ros/jazzy/share/ur_robot_driver/config/ur_controllers.yaml` to change the joint names specified for the `joint_trajectory_controller`. (e.g. add an `_a` suffix to one of the joints and its constraints)
7. Rerun the above ros2 launch command. It should print a stacktrace similar to the following:

```
[ros2_control_node-1] [INFO] [1776672277.141229928] [controller_manager]: Loading controller 'joint_trajectory_controller'
[ros2_control_node-1] [INFO] [1776672277.141542621] [controller_manager]: Controller 'joint_trajectory_controller' node arguments: --ros-args --params-file /opt/ros/jazzy/share/ur_robot_driver/config/ur10e_update_rate.yaml --params-file /tmp/launch_params_xoei9ciw 
[ros2_control_node-1] Stack trace (most recent call last) in thread 4726:
[ros2_control_node-1] #15   Object "", at 0xffffffffffffffff, in 
[ros2_control_node-1] #14   Object "/usr/lib/x86_64-linux-gnu/libc.so.6", at 0x78d143417a63, in __clone
[ros2_control_node-1] #13   Object "/usr/lib/x86_64-linux-gnu/libc.so.6", at 0x78d14338aaa3, in 
[ros2_control_node-1] #12   Object "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.33", at 0x78d14361adb3, in 
[ros2_control_node-1] #11   Object "/opt/ros/jazzy/lib/librclcpp.so", at 0x78d1438df9c6, in rclcpp::executors::MultiThreadedExecutor::run(unsigned long)
[ros2_control_node-1] #10   Object "/opt/ros/jazzy/lib/librclcpp.so", at 0x78d1438ccf99, in rclcpp::Executor::execute_any_executable(rclcpp::AnyExecutable&)
[ros2_control_node-1] #9    Object "/opt/ros/jazzy/lib/librclcpp.so", at 0x78d1438c8d33, in rclcpp::Executor::execute_service(std::shared_ptr<rclcpp::ServiceBase>)
[ros2_control_node-1] #8    Object "/opt/ros/jazzy/lib/librclcpp.so", at 0x78d1439a27f9, in 
[ros2_control_node-1] #7    Object "/opt/ros/jazzy/lib/libcontroller_manager.so", at 0x78d143b8a5fd, in 
[ros2_control_node-1] #6    Object "/opt/ros/jazzy/lib/libcontroller_manager.so", at 0x78d143b6b739, in std::_Function_handler<void (std::shared_ptr<controller_manager_msgs::srv::LoadController_Request_<std::allocator<void> > >, std::shared_ptr<controller_manager_msgs::srv::LoadController_Response_<std::allocator<void> > >), std::_Bind<void (controller_manager::ControllerManager::*(controller_manager::ControllerManager*, std::_Placeholder<1>, std::_Placeholder<2>))(std::shared_ptr<controller_manager_msgs::srv::LoadController_Request_<std::allocator<void> > >, std::shared_ptr<controller_manager_msgs::srv::LoadController_Response_<std::allocator<void> > >)> >::_M_invoke(std::_Any_data const&, std::shared_ptr<controller_manager_msgs::srv::LoadController_Request_<std::allocator<void> > >&&, std::shared_ptr<controller_manager_msgs::srv::LoadController_Response_<std::allocator<void> > >&&)
[ros2_control_node-1] #5    Object "/opt/ros/jazzy/lib/libcontroller_manager.so", at 0x78d143b10d07, in controller_manager::ControllerManager::load_controller_service_cb(std::shared_ptr<controller_manager_msgs::srv::LoadController_Request_<std::allocator<void> > >, std::shared_ptr<controller_manager_msgs::srv::LoadController_Response_<std::allocator<void> > >)
[ros2_control_node-1] #4    Object "/opt/ros/jazzy/lib/libcontroller_manager.so", at 0x78d143aff553, in controller_manager::ControllerManager::load_controller(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
[ros2_control_node-1] #3    Object "/opt/ros/jazzy/lib/libcontroller_manager.so", at 0x78d143afe213, in controller_manager::ControllerManager::load_controller(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
[ros2_control_node-1] #2    Object "/opt/ros/jazzy/lib/libcontroller_manager.so", at 0x78d143b0c6bd, in controller_manager::ControllerManager::add_controller_impl(controller_manager::ControllerSpec const&)
[ros2_control_node-1] #1    Object "/opt/ros/jazzy/lib/libcontroller_interface.so", at 0x78d1432221ee, in controller_interface::ControllerInterfaceBase::init(controller_interface::ControllerInterfaceParams const&)
[ros2_control_node-1] #0    Object "/opt/ros/jazzy/lib/libjoint_trajectory_controller.so", at 0x78d11817c08f, in joint_trajectory_controller::JointTrajectoryController::on_init()
[ros2_control_node-1] Segmentation fault (Address not mapped to object [0xc8])
[ERROR] [ros2_control_node-1]: process has died [pid 4629, exit code -11, cmd '/opt/ros/jazzy/lib/controller_manager/ros2_control_node --ros-args --params-file /opt/ros/jazzy/share/ur_robot_driver/config/ur10e_update_rate.yaml --params-file /tmp/launch_params_xoei9ciw'].
```

By building from scratch (and with `RelWithDebInfo`), we get a more useful stacktrace:

```
...
[ros2_control_node-1]       >1315:   return add_controller_impl(controller_spec);
[ros2_control_node-1]        1316: }
[ros2_control_node-1]        1317: 
[ros2_control_node-1]        1318: controller_interface::ControllerInterfaceBaseSharedPtr ControllerManager::load_controller(
[ros2_control_node-1] #2    Source "/workspace/src/ros2_control/controller_manager/src/controller_manager.cpp", line 2409, in add_controller_impl [0x729dc248dacc]
[ros2_control_node-1]        2406:     controller_params.node_options = controller_node_options;
[ros2_control_node-1]        2407:     controller_params.hard_joint_limits = resource_manager_->get_hard_joint_limits();
[ros2_control_node-1]        2408:     controller_params.soft_joint_limits = resource_manager_->get_soft_joint_limits();
[ros2_control_node-1]       >2409:     if (controller.c->init(controller_params) == controller_interface::return_type::ERROR)
[ros2_control_node-1]        2410:     {
[ros2_control_node-1]        2411:       to.clear();
[ros2_control_node-1]        2412:       RCLCPP_ERROR(
[ros2_control_node-1] #1    Source "/workspace/src/ros2_control/controller_interface/src/controller_interface_base.cpp", line 170, in init [0x729dc1b6d331]
[ros2_control_node-1]         167:       return transition_state_status;
[ros2_control_node-1]         168:     });
[ros2_control_node-1]         169: 
[ros2_control_node-1]       > 170:   switch (on_init())
[ros2_control_node-1]         171:   {
[ros2_control_node-1]         172:     case LifecycleNodeInterface::CallbackReturn::SUCCESS:
[ros2_control_node-1]         173:       break;
[ros2_control_node-1] #0    Source "/workspace/src/ros2_controllers/joint_trajectory_controller/src/joint_trajectory_controller.cpp", line 83, in on_init [0x729da0177b48]
[ros2_control_node-1]          80:       for (size_t i = 0; i < params_.joints.size(); ++i)
[ros2_control_node-1]          81:       {
[ros2_control_node-1]          82:         auto urdf_joint = model.getJoint(params_.joints[i]);
[ros2_control_node-1]       >  83:         max_joint_vel[i] = urdf_joint->limits->velocity;
[ros2_control_node-1]          84:         if (urdf_joint && urdf_joint->type == urdf::Joint::CONTINUOUS)
[ros2_control_node-1]          85:         {
[ros2_control_node-1]          86:           RCLCPP_DEBUG(
[ros2_control_node-1] Segmentation fault (Address not mapped to object [0xc8])
```

Since the joint is not found, `urdf_joint` is null, so [line 83](https://github.com/ros-controls/ros2_controllers/blob/9e7dc558404ef98de7191af68b7922d07d358f67/joint_trajectory_controller/src/joint_trajectory_controller.cpp#L83) segfaults. The check on line 84 (or at least the first part of the check) could have prevented this.

**Expected behavior**
Initialisation should succeed regardless of whether a joint name is in the urdf or not. As per the following comment:
https://github.com/ros-controls/ros2_controllers/blob/9e7dc558404ef98de7191af68b7922d07d358f67/joint_trajectory_controller/src/joint_trajectory_controller.cpp#L91

Note: I am not sure what the default `max_joint_vel` should be set for if the joint does not exist. Setting it negative/equal to zero could work, but would disable decelerate-on-cancel.

It shouldn't be too hard to also add this to the tests, so these bugs are caught automatically in the future.

**Environment (please complete the following information):**
 - OS: Ubuntu 24.04 Noble
 - Version Jazzy (`4.39.0`, as reported by `ros2 pkg xml -t version joint_trajectory_controller`)

**Additional context**
This bug may also be present in the `kilted` and `rolling` versions.

--- a/joint_trajectory_controller/src/joint_trajectory_controller.cpp
+++ b/joint_trajectory_controller/src/joint_trajectory_controller.cpp
@@ -80,7 +80,9 @@ controller_interface::CallbackReturn JointTrajectoryController::on_init()
       for (size_t i = 0; i < params_.joints.size(); ++i)
       {
         auto urdf_joint = model.getJoint(params_.joints[i]);
-        max_joint_vel[i] = urdf_joint->limits->velocity;
+        if (urdf_joint){
+          max_joint_vel[i] = urdf_joint->limits->velocity;
+        }
         if (urdf_joint && urdf_joint->type == urdf::Joint::CONTINUOUS)
         {
           RCLCPP_DEBUG(

--- a/joint_trajectory_controller/src/joint_trajectory_controller.cpp
+++ b/joint_trajectory_controller/src/joint_trajectory_controller.cpp
@@ -80,7 +80,8 @@ controller_interface::CallbackReturn JointTrajectoryController::on_init()
       for (size_t i = 0; i < params_.joints.size(); ++i)
       {
         auto urdf_joint = model.getJoint(params_.joints[i]);
-        if (urdf_joint){
+        if (urdf_joint)
+        {
           max_joint_vel[i] = urdf_joint->limits->velocity;
         }
         if (urdf_joint && urdf_joint->type == urdf::Joint::CONTINUOUS)

